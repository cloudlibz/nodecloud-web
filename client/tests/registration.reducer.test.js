import * as reducer from "../src/_reducers/registration.reducer";

describe("reducer", () => {
  describe("REGISTRATION REQUEST", () => {
    test("sets registration to requesting", function() {
      const previousState = {
        otherProp: "test-123"
      };
      const action = {
        type: "USERS_REGISTER_REQUEST"
      };
      const nextState = reducer.registration(previousState, action);
      expect(nextState).toMatchSnapshot();
    });
  });

  describe("REGISTRATION SUCCESS", () => {
    test("sets registration to success", function() {
      const previousState = {
        otherProp: "test-123"
      };
      const action = {
        type: "USERS_REGISTER_SUCCESS"
      };
      const nextState = reducer.registration(previousState, action);
      expect(nextState).toMatchSnapshot();
    });
  });

  describe("REGISTRATION FAILURE", () => {
    test("sets registration to failure", function() {
      const previousState = {
        otherProp: "test-123"
      };
      const action = {
        type: "USERS_REGISTER_FAILURE"
      };
      const nextState = reducer.registration(previousState, action);
      expect(nextState).toMatchSnapshot();
    });
  });
});
