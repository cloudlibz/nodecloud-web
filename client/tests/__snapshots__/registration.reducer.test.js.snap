// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer REGISTRATION FAILURE sets registration to failure 1`] = `Object {}`;

exports[`reducer REGISTRATION REQUEST sets registration to requesting 1`] = `
Object {
  "registering": true,
}
`;

exports[`reducer REGISTRATION SUCCESS sets registration to success 1`] = `Object {}`;
